tools-local/doc.tasks/GenerateMarkdownDoc.cs

```diff
@@ -90,14 +94,62 @@ public override bool Execute()
         private void WriteSectionProjecs(ITaskItem section, StreamWriter file)
         {
             string[] projectFiles = section.GetMetadata("Projects").Split(';');
+            var separator = Path.DirectorySeparatorChar;
 
-            file.WriteLine($"Projects in section: {projectFiles.Length}\n");
-                    
+            file.WriteLine($"Projects in section: {projectFiles.Length}");
+
+            //Logic below relies on pre-sorting to identify group/folder structure changes.
             Array.Sort(projectFiles, (a, b) => a.CompareTo(b));
 
+            string prevGroupName = null;
+
             for (int i = 0; i < projectFiles.Length; i++)
             {
+                string groupName = null;
+                string fileName = null;
+
                 var projectPath = projectFiles[i];
+                var relativeProjectPath = RelativizePath(projectPath);
+                var pathSplit = relativeProjectPath.Split(separator);
+
+                if (pathSplit == null || pathSplit.Length < 1)
+                {
+                    continue;
+                }
+                else if (pathSplit.Length == 2)
+                {
+                    groupName = pathSplit[0];
+                    fileName = pathSplit[1];
+                }
+                else //More than 2 subfolders will be grouped by [subfolders] => [project name].
+                {
+                    var end = pathSplit.Length - 1;
+                    fileName = pathSplit[end];
+
+                    var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
+                    if (end > 0)
+                    {
+                        //Ignore the last subfolder if its name matches the filename.
+                        if (pathSplit[end - 1].Equals(fileNameWithoutExtension, StringComparison.OrdinalIgnoreCase))
+                        {
+                            --end;
+                        }
+
+                        groupName = string.Join(separator.ToString(), pathSplit.Take(end));
+                    }
+                    else
+                    {
+                        groupName = pathSplit[0];
```

Comment:
Imagine a case in which we have the following paths (already sorted):

```
a.csproj
b.csproj
c\091\u.csproj
c\ab\x.csproj
z.csproj
```
I imagine that it will create a group for each file not in a folder (a, b, z)

A workaround is to sort files windows-explorer style; first files, then folders

Reply:
In my opinion, files without a folder should not be in a group

Reply:
Root-level csproj's will appear as separate groups. Discussed offline that this is acceptable, and since we didn't come up with how we would want it handled, and it shouldn't happen in this repo.